### DISCUSSION TITLE
MenuItem disabled if command property is present.

### DISCUSSION THREAD

**Discussion by michalss** [AvaloniaUI/Avalonia#13819] [Q&A]
## Describe the bug
When i use command to MenuItem its become unusable. It simply grays out and disable, where i cannot click on it.

## To Reproduce
Make a Avalonia project add MenuItem and use CommunityToolkit.Mvvm for binding.

![image](https://github.com/AvaloniaUI/Avalonia/assets/1000729/526f1b57-3466-4661-8a07-71495295da67)

![image](https://github.com/AvaloniaUI/Avalonia/assets/1000729/1c90c4c7-2b15-40a2-8bb8-d086dc994735)

## Expected behavior
Well i would like to use commands on MenuItems, so it should simply work as expected like buttons..

## Screenshots
![image](https://github.com/AvaloniaUI/Avalonia/assets/1000729/64201c23-98df-497f-b19f-71e8f2d5dd41)

## Environment 
Win11 , Vs2022

**Comment by Coloryr** [comment]
try `{Binding RunScript}`
and set method public

**Comment by Alshain01** [comment]
> try `{Binding RunScript}` and set method public

No, it would be RunScriptCommand in community toolkit with code generators and the backing method can be private, the generated Command property will still be public.  I know Avalonia supports direct binding to methods but that's not what he is doing here.  He is binding to an ICommand that is autogenerated in a separate partial class.

**Comment by michalss** [comment]
> > try `{Binding RunScript}` and set method public
> 
> No, it would be RunScriptCommand in community toolkit with code generators and the backing method can be private, the generated Command property will still be public. I know Avalonia supports direct binding to methods but that's not what he is doing here. He is binding to an ICommand that is autogenerated in a separate partial class.

Yes this is pretty common for CommunityToolkit. So is there anything i can do ?

**Comment by Coloryr** [comment]
![1](https://github.com/AvaloniaUI/Avalonia/assets/26276037/f6520869-a6c9-4561-ab2b-e4946b200349)
Maybe your `DataContext` is incorrect

**Comment by michalss** [comment]
> ![1](https://private-user-images.githubusercontent.com/26276037/287611209-f6520869-a6c9-4561-ab2b-e4946b200349.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2NzM0NjIsIm5iZiI6MTcwMTY3MzE2MiwicGF0aCI6Ii8yNjI3NjAzNy8yODc2MTEyMDktZjY1MjA4NjktYTZjOS00NTYxLWFiMmItZTQ5NDZiMjAwMzQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA0VDA2NTkyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhOTM4OWI0ZWJjY2QzNDNhNDEyYmRhNjQwOTRkYWM5ZDk1NTg3NTQzYjhjYTRiNTJmYmZjYTQzMDExNDgzMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JCrNG-oj6Qfj3glc6VIiOjz2CbY_UofNSnYTF0FzZ4I) Maybe your `DataContext` is incorrect

it is very strange, only different i can see is im using  UserControl instead of Window.. hmm.. Can you please pack it and send it over, mean that sample you made. I would like to compare also versions etc... ?

**Comment by Coloryr** [comment]
> is is very strange, only different i can see is im using UserControl instead of Window.. hmm.. Can you please pack it and send it over, mean that sample you made. I would like to compare also versions etc... ?

[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/13542891/AvaloniaApplication1.zip)

**Comment by timunie** [comment]
@michalss try to inspect you DataContext with DevTools [F12]. If you have compiled bindings and the command isn't found for some reason this happens. If that still doesn't help please send over a minimal sample to reproduce. 

Also which Avalonia version do you use?

**Comment by michalss** [comment]
i can confirm it is a bug.. here is project . If you add MenuItem under UserControl, then commands does not work. To be perfectly clear, if use command with MenuItem, that button will be disabled completely.

[TestMVVMBug.zip](https://github.com/AvaloniaUI/Avalonia/files/13543544/TestMVVMBug.zip)

```csharp
    <PackageReference Include="Avalonia" Version="11.0.5" />
    <PackageReference Include="Avalonia.AvaloniaEdit" Version="11.0.5" />
    <PackageReference Include="Avalonia.Desktop" Version="11.0.5" />
    <PackageReference Include="Avalonia.Svg.Skia" Version="11.0.0.6" />
    <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.5" />
    <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.5" />
    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.2" />
```

**Comment by Coloryr** [comment]
> i can confirm it is a bug.. here is project . If you add MenuItem under UserControl, then commands does not work. To be perfectly clear, if use command with MenuItem, that button will be disabled completely.
> 
> [TestMVVMBug.zip](https://github.com/AvaloniaUI/Avalonia/files/13543544/TestMVVMBug.zip)
> 
> ```cs
>     <PackageReference Include="Avalonia" Version="11.0.5" />
>     <PackageReference Include="Avalonia.AvaloniaEdit" Version="11.0.5" />
>     <PackageReference Include="Avalonia.Desktop" Version="11.0.5" />
>     <PackageReference Include="Avalonia.Svg.Skia" Version="11.0.0.6" />
>     <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.5" />
>     <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.5" />
>     <PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.2" />
> ```

your DataContext not set 

![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/b001b2cf-7527-4989-ba77-3bd54062233d)

Here is the correct code

[TestMVVMBug (2).zip](https://github.com/AvaloniaUI/Avalonia/files/13544256/TestMVVMBug.2.zip)

**Comment by timunie** [comment]
Your sample has issues. If I correct them, the sample works as intended: 

```diff
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
			 xmlns:vm="using:TestMVVMBug.ViewModels"
!                        // DataType was wrong. You have MainWindowViewModel and MainViewModel. However, DataContext is MainWindowViewModel inherited from the parent. 
!			 x:DataType="vm:MainWindowViewModel"
			 xmlns:viewModel="clr-namespace:TestMVVMBug.ViewModels;assembly=TestMVVMBug"
             x:Class="TestMVVMBug.Views.MainView">

	<Design.DataContext>
!               // This was correct already, so you didn't see it in design time
		<viewModel:MainWindowViewModel/>
	</Design.DataContext>

	<Grid RowDefinitions="20,*" Margin="10">
		<Menu Grid.Row="0">
			<MenuItem  VerticalAlignment="Top" Header="_Scripting">
!                               // Command name was wrong in your sample
!				<MenuItem x:Name="RunScript" Header="_Run Script" Command="{Binding RunScriptmainVidowCommand}" />
			</MenuItem>

		</Menu>
		<Grid Grid.Row="1" VerticalAlignment="Center" HorizontalAlignment="Center">
			<TextBlock>
				Contennt
			</TextBlock>
		</Grid>
	</Grid>

</UserControl>
```

Note. As the issue is not reproducible for us, I'll convert it into a discussion.

**Comment by michalss** [comment]
ok i got it it seems that problem is Dock.Avalonia :( not avalonia itself.. :(
